### PR TITLE
OP: enable Resize/Prelu/Sigmoid, modified Reshape

### PR DESCRIPTION
**Pull request by erizmr**
Hi @jackwish , 

The main contributions in this PR are listed below:
1. Enable three new operators: `Resize`, `Prelu`, `Sigmoid`(seems that Sigmoid has been developed in another pull request, sorry for this redundant work..) 
2. Modified `Reshape` to adapt cases with one attribute
3. Produce test models for three new operators (seems the model files are list in the .gitignore, so I excluded them from this PR )

Thanks for your awesome code review in advance :)

### CODE REVIEW

**Comment by jackwish** [comment]
Hi @erizmr thanks for your contribution! It seems you forget to include the TFLite model file of `resize_bilinear` in this PR :)

**Comment by jackwish** [comment]
For adding TFLite model files, you may need to use `git add -f {file}` as `*.tflite` is ignored by default.

**Comment by erizmr** [comment]
Thanks! Have uploaded the missing models.

**Review by jackwish — CHANGES_REQUESTED** [review]
@erizmr thank you for your great contribution! I have some minor comments, and I will read the `Resize` part tomorrow.

Btw, would you mind helping to update the [Operator Support Status](docs/operator-support-status.md) to demonstrate the new ops in this PR?

File: tflite2onnx/op/activation.py [hunk 1]
```diff
@@ -1,15 +1,51 @@
...
 
+from tflite2onnx import mapping
 from tflite2onnx.op.common import Operator
 
 logger = logging.getLogger('tflite2onnx')
 
 
+class Logistic(Operator):
+    TypeMapping = {
+        tflite.BuiltinOperator.LOGISTIC: 'Sigmoid',
```
[the remark this answers is not in the document]
Comment (erizmr) [reply tflite2onnx/op/activation.py:12]: Sure. Happy to update it.

File: tflite2onnx/op/activation.py [hunk 2]
```diff
@@ -53,6 +92,15 @@ def parse(self):
             tmax.addConsumer(self)
             self.inputs.append(tmax)
 
+        if opcode == tflite.BuiltinOperator.PRELU:
+            # `alpha` should be a learned array with the same shape as `X`
+            # But there is no `batch_size` dimension in its shape,
+            # which will cause `out of index` exception during axis transform
+            # so we copy the `batch_size` from `X` to expand its dimension
+            alpha = self.parseInput(1)
+            alpha.shape.insert(0, self.inputs[0].shape[0])
```
[the remark this answers is not in the document]
Comment (erizmr) [reply tflite2onnx/op/activation.py:101]: Agree. `1` generalizes better.
There is an argument `alpha_initializer` that help initialize the values, I will try to generate a new one.

File: tflite2onnx/op/reshape.py [hunk 3]
```diff
@@ -32,24 +32,38 @@ def parse(self):
...
-        if st.isInitializer:
-            st.data = st.data.astype('int64')
-        if (len(st.shape) > 1):
-            logger.warning("ONNXRuntime doesn't support 2+rank shape, "
-                           "flatten if the shape is initialzier, ignore otherwise."
-                           "https://github.com/jackwish/tflite2onnx/issues/16")
+        data = self.parseInput(0)
+
+        if op.InputsLength() == 1:
+            # options
```
[the remark this answers is not in the document]
Comment (erizmr) [reply tflite2onnx/op/reshape.py:42]: That's the interesting thing. Both `tf.reshape` and 'tf.keras.layers.Reshape' output two-inputs blocks. Some models from [mediapipe](https://github.com/google/mediapipe/blob/master/mediapipe/modules/palm_detection/palm_detection.tflite) do have such one-input `Reshape` operators. I can try to figure it out.

File: tflite2onnx/op/resize.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+from tflite2onnx.layout import Layout
+from tflite2onnx import mapping
+from tflite2onnx.op.common import Operator
+
+logger = logging.getLogger('tflite2onnx')
+
+
+class Resize(Operator):
+    TypeMapping = {
+        tflite.BuiltinOperator.RESIZE_NEAREST_NEIGHBOR: 'Resize',
```
Comment (jackwish) [inline tflite2onnx/op/resize.py:14]: As we are enabling `RESIZE_NEAREST_NEIGHBOR` here too, it would be great to have a dedicated TFLite model for test.

File: tflite2onnx/op/resize.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        return opcode is tflite.BuiltinOperator.RESIZE_NEAREST_NEIGHBOR
+
+    def parse(self):
+        logger.debug("Parsing %s...", self.type)
+        op = self.tflite
+        opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+        assert(opcode is tflite.BuiltinOperator.RESIZE_BILINEAR
+               or tflite.BuiltinOperator.RESIZE_NEAREST_NEIGHBOR)
+
+        assert(op.InputsLength() >= 2), "Only first two arguments: image and size, are compulsory"
```
Comment (jackwish) [inline tflite2onnx/op/resize.py:70]: I checked TFLite's kernel, both `resize_bilinear` and `resize_nearest_neighbor` require two inputs. Have we met scenarios against this?

File: tflite2onnx/op/resize.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        # of ResizeBilinear and ResizeNearestNeighbor
+        holder = self.TFactory.createVector(im, np.array([]))
+        holder.parse()
+        holder.addConsumer(self)
+        self.inputs.append(holder)
+
+        # Fill empty to scales temporarily
+        sc = self.TFactory.createVector(im, np.array([]))
+        sc.parse()
+        self.inputs.append(sc)
```
Comment (jackwish) [inline tflite2onnx/op/resize.py:88]: Do we need to add `Resize` as a consumer of `sc`?

File: tflite2onnx/op/resize.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        assert(op.OutputsLength() == 1)
+
+        # image
+        ilayout = Layout('NHWC', 'NCHW')
+        im = self.parseInput(0, ilayout)
+
+        # Fill 'ROI' empty temporarily
+        # because 'tf_crop_and_resize' was not found in BuiltinOptions
+        # of ResizeBilinear and ResizeNearestNeighbor
+        holder = self.TFactory.createVector(im, np.array([]))
```
Comment (jackwish) [inline tflite2onnx/op/resize.py:80]: Maybe we can just name it as `roi`?

**Review by jackwish — CHANGES_REQUESTED** [review]
@erizmr Thanks for your active work on this :) I have quick look at the `Resize` part and left several comments, but I need some time to check the interpolation in detail still.

The Codecov requires coverage don't drop > 1% (I am not sure if it's correctly configured though...), we may try to get that.

File: tflite2onnx/op/resize.py (hunk 6 above)
Comment (ashikns) [reply tflite2onnx/op/resize.py:88]: Sorry to jump in, but I happened to be in need of Resize myself when I found this.

Adding self as a consumer of sc fails because "sc" here returned is the same tensor object from previous line, i.e. "holder". This happens because both of their names resolve to the same. I'm guessing they need to be forced to be unique somehow?

Comment (jackwish) [reply tflite2onnx/op/resize.py:88]: That makes sense, we may use `holder` directly currently. And, welcome @ashikns :)

**Comment by erizmr** [comment]
Hi @jackwish , thanks for your review! Please check again.

File: tflite2onnx/op/resize.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        op = self.tflite
+        opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+        assert(opcode is tflite.BuiltinOperator.RESIZE_BILINEAR
+               or tflite.BuiltinOperator.RESIZE_NEAREST_NEIGHBOR)
+
+        assert(op.InputsLength() >= 2), "Only first two arguments: image and size, are compulsory"
+        assert(op.OutputsLength() == 1)
+
+        # image
+        ilayout = Layout('NHWC', 'NCHW')
```
[the remark this answers is not in the document]
Comment (jackwish) [reply tflite2onnx/op/resize.py:74]: https://github.com/jackwish/tflite2onnx/issues/26#issue-740792901

**Review by jackwish — APPROVED** [review]
@erizmr LGTM in general, thanks for your contribution!
